docs/proposals/container-init.md

```diff
@@ -0,0 +1,333 @@
+            containers:
+            - name: run
+              image: application-image
+              command: ["/my_application_that_depends_on_myservice"]
+
+* Wait for an arbitrary period of time
+
+        pod:
+          spec:
+            initContainers:
+            - name: wait
+              image: centos:centos7
+              command: ["/bin/sh", "-c", "sleep 60"]
+            containers:
+            - name: run
+              image: application-image
+              command: ["/static_binary_without_sleep"]
+
+* Clone a git repository into a volume:
+
+        pod:
+          spec:
+            initContainers:
+            - name: download
+              image: image-with-git
+              command: ["git", "clone", "https://github.com/myrepo/myrepo.git", "/var/lib/data"]
+              volumeMounts:
+              - mountPath: /var/lib/data
+                volumeName: git
+            containers:
+            - name: run
+              image: centos:centos7
+              command: ["/var/lib/data/binary"]
+              volumeMounts:
+              - mountPath: /var/lib/data
+                volumeName: git
+            volumes:
+            - emptyDir: {}
+              name: git
+
+* Execute a template transformation based on environment
+
+        pod:
+          spec:
+            initContainers:
+            - name: copy
+              image: application-image
+              command: ["/bin/cp", "mytemplate.j2", "/var/lib/data/"]
+              volumeMounts:
+              - mountPath: /var/lib/data
+                volumeName: data
+            - name: transform
+              image: image-with-jinja
+              command: ["/bin/sh", "-c", "jinja /var/lib/data/mytemplate.j2 > /var/lib/data/mytemplate.conf"]
+              volumeMounts:
+              - mountPath: /var/lib/data
+                volumeName: data
+            containers:
+            - name: run
+              image: application-image
+              command: ["/myapplication", "-conf", "/var/lib/data/mytemplate.conf"]
+              volumeMounts:
+              - mountPath: /var/lib/data
+                volumeName: data
+            volumes:
+            - emptyDir: {}
+              name: data
+
+* Perform a container build
+
+        pod:
+          spec:
+            initContainers:
+            - name: copy
+              image: base-image
+              workingDir: /home/user/source-tree
+              command: ["make"]
+            containers:
+            - name: commit
+              image: image-with-docker
+              command:
+              - /bin/sh
+              - -c
+              - docker commit $(complex_bash_to_get_container_id_of_copy) \
+                docker push $(commit_id) myrepo:latest
+              volumesMounts:
+              - mountPath: /var/run/docker.sock
+                volumeName: dockersocket
+
+
+## Open Questions
+
+* We could support readiness on init containers, in which case there is no difference
```

Comment:
Would #831 obviate the need for init containers? Expressing init container that don't generate any config data with volume containers might be clunky, but other than that, are there any other use cases that cannot be captured using volume containers?